cpp/include/cugraph/mtmg/detail/per_device_edgelist.hpp

```diff
@@ -110,45 +125,67 @@ class per_device_edgelist_t {
               std::optional<raft::host_span<edge_t const>> edge_id,
               std::optional<raft::host_span<edge_type_t const>> edge_type)
   {
-    // FIXME:  This lock guard could be on a smaller region, but it
-    //   would require more careful coding.  The raft::update_device
-    //   calls could be done without the lock if we made a local
-    //   of the values of *.back() and did an increment of current_pos_
-    //   while we hold the lock.
-    std::lock_guard<std::mutex> lock(lock_);
+    std::vector<std::tuple<vertex_t*, vertex_t const*, size_t>> src_copies;
+    std::vector<std::tuple<vertex_t*, vertex_t const*, size_t>> dst_copies;
+    std::vector<std::tuple<weight_t*, weight_t const*, size_t>> wgt_copies;
+    std::vector<std::tuple<edge_t*, edge_t const*, size_t>> edge_id_copies;
+    std::vector<std::tuple<edge_type_t*, edge_type_t const*, size_t>> edge_type_copies;
```

Comment:
Should we maintain 5 variables or one variable storing (input_start_offset, output_start_offset, size) triplets will be sufficient?

Reply:
Good suggestion, I'll look into that for next push.